build.sbt

```diff
@@ -186,14 +186,15 @@ val commonSettings = Seq(
   tlFatalWarningsInCi := false,
   tlJdkRelease := Some(8),
   tlSkipIrrelevantScalas := true,
+  scalacOptions ~= { _.filterNot(_ == "-source:3.0-migration") },
```

Comment:
can you explain what this option means?

Reply:
`source:3.0-migration` instructs the scala 3 compiler to run in [migration mode](https://docs.scala-lang.org/scala3/guides/migration/tooling-migration-mode.html). This is the default from `sbt-typlevel` when cross-building with scala 2 (the sources are the same).
However, with this option, the compiler fails on the dedicated `scala-3` folder because sources are not in the scala 2 style. See discussion https://github.com/lampepfl/dotty/discussions/16720